### PR TITLE
app-doc/phrack: Fix build for slots with a leading zero

### PR DESCRIPTION
**Pull request by holgersson32644**
This commit bumps all ebuilds at once. Porting the change to newer ebuilds makes no sense
as a) they won't start with a leading zero, and b) they probably won't increase anyway.

### CODE REVIEW

**Comment by holgersson32644** [comment]
Closed via https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=9663d70b415c329ec1f634f402af3e8f2e3cc67d.

**Comment by holgersson32644** [comment]
Correction: No, this fixes not the actual issue (wrong ${S} (just the EAPI bump).

File: app-doc/phrack/phrack-01-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# fix PV for slots starting with a 0
+# see https://bugs.gentoo.org/653418
+if [[ ${PV} < 10 ]]
```
Comment (mgorny) [inline app-doc/phrack/phrack-01-r1.ebuild]: `<` is string comparison, not integer.

File: app-doc/phrack/phrack-01-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+then
+	MY_P="${PN}0${PV:1:1}"
+else
+	MY_P="${PN}${PV:1:2}"
+fi
+
+DESCRIPTION="A Hacker magazine by the community, for the community"
+HOMEPAGE="http://www.phrack.org/"
+SRC_URI="http://www.phrack.org/archives/tgz/${MY_P}.tar.gz"
```
Comment (mgorny) [inline app-doc/phrack/phrack-01-r1.ebuild]: This SRC_URI is invalid. You're reverting the earlier fix.

File: app-doc/phrack/phrack-01-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# fix PV for slots starting with a 0
+# see https://bugs.gentoo.org/653418
+if [[ ${PV} < 10 ]]
+then
+	MY_P="${PN}0${PV:1:1}"
```
Comment (mgorny) [inline app-doc/phrack/phrack-01-r1.ebuild]: So… you're stripping the zero… just to prepend the zero?

File: app-doc/phrack/phrack-01-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+DESCRIPTION="A Hacker magazine by the community, for the community"
+HOMEPAGE="http://www.phrack.org/"
+SRC_URI="http://www.phrack.org/archives/tgz/${MY_P}.tar.gz"
+
+LICENSE="phrack"
+SLOT="${PV}"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+IUSE=""
+
+S="${WORKDIR}/${MY_P}"
```
Comment (mgorny) [inline app-doc/phrack/phrack-01-r1.ebuild]: This is incorrect for all those versions.

**Comment by mgorny** [comment]
Since apparently the only change that needs to happen is setting `S=${WORKDIR}` in 10 ebuilds, I'll just handle it. Thanks for pinging us about the problem.

**Comment by holgersson32644** [comment]
Am Freitag, 27. Juli 2018, 15:12:37 CEST schrieb Larry the Cow:
> Closed #9138 via dccd17ca6b055ffdc0f5f2239563f177775e9e08.
Fine, thanks!

**Pull request closed without merging**